src/k8s/pkg/k8sd/database/cluster_configs.go

```diff
@@ -0,0 +1,112 @@
+package database
+
+import (
+	"context"
+	"database/sql"
+	_ "embed"
+	"fmt"
+
+	"github.com/canonical/microcluster/cluster"
+	"github.com/mitchellh/mapstructure"
+)
+
+var (
+	clusterConfigsStmts = map[string]int{
+		"upsert-config": mustPrepareStatement("cluster-configs", "upsert-config.sql"),
+		"select-all":    mustPrepareStatement("cluster-configs", "select-all.sql"),
+	}
+)
+
+type ClusterConfig struct {
+	// K8sCaCert represents the Kubernetes Certificate Authority certificate.
+	// Empty if we don't use self-signed certificates.
+	K8sCaCert string `mapstructure:"k8s-ca-crt,omitempty"`
```

Comment:
Let's avoid abbreviations on struct field members, so that it's easier to track their use across the codebase.

e.g. instead I would recommend


	K8sCertificateAuthority string `mapstructure:"k8s-ca-crt,omitempty"`

	K8sCertificateAuthorityKey string `mapstructure:"k8s-ca-crt,omitempty"`

 Also, for apiserver configs, prefer `APIServerRBAC` capitalization.

Reply:
The intention was to have a 1:1 mapping between db key and member.
But I see the problem with it - let's make it more explicit.